packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -145,23 +143,18 @@ function getAllAttributesInRelation({ relations }: Relations): Attributes[] {
 
 // Function checks if attribute is valid and has location data in it.
 // Since have observed noisy data before so this function is required.
-function isValidAttribute(attr: Attributes): boolean {
-  if (attr && attr.location && attr.location.begin && attr.location.end) {
-    return true;
-  }
-  return false;
+function hasLocationData(attr: Attributes): boolean {
+  return !!(attr && attr.location && attr.location.begin && attr.location.end);
 }
 
 export default function transformEnrichment(enrichedHtml: EnrichedHtml[]): EnrichedHtml[] {
-  const newEnrichedHtml = { ...enrichedHtml };
-
-  if (newEnrichedHtml && newEnrichedHtml[0]) {
-    const enrichmentName = getEnrichmentName(newEnrichedHtml[0]);
-    const enrichment = newEnrichedHtml[0][enrichmentName];
+  if (enrichedHtml && enrichedHtml[0]) {
```

Comment:
If you're giving `enrichedHtml` the type `EnrichedHtml[]`, then do you still need to check if it is defined? I'd think that typescript would throw an error for anyone using this method if they tried to pass in a variable that could be `undefined` or `EnrichedHtml[]`, and if they ignore that error and this fails, that's their fault.

Reply:
I kept it like this for additional failsafe.